### DISCUSSION TITLE
A little tweak for Foragers

### DISCUSSION THREAD

**Discussion by ADC021499** [Courseplay/Courseplay_FS22#2381] [Ideas]
### A little background.

Partly for learning using CP + AD together, and largely for stress testing, I loaded up American Farmlands, bought the entire SE quarter of the map, and made a master field for CP that included the entire square, minus the small "field" that encompasses the Depot and Animal Sell point.  I set up three 10.5 million liter silos on this space, named "Silo Red", "Silo Green" and "Silo Blue".  Then 9 tractors, 3 red, 3 green, 3 blue, and gave them all trailers.

Another 3 tractors (these black) for doing the preparatory work of mowing (and in one experiment, tedding and windrowing).

My original intent was to use a forager, but I created a field about one third the size of the main field, and created a CP route to turn the grass, thinking I'd want hay for my dairy farm.  So I mowed the entire main field, then ran three tedders over the subset "Hay" field, then 3 windrowers over the main field as a whole.

### The Experiments.

**The first test** 
...came to an early end when I saw my foragers were not picking up the hay.  I'm still kinda suprised there are no headers that will collect hay or straw, but that's another discussion.  So instead of foragers, I supplied my 9 colored tractors with foraging trailers (which happily pick grass, hay, and straw).

They ran the same course as the windrowers; a 3 tool, 1 headland course.  Center start.

And yes, it was actually successful.  One thing that would make this a little faster would be if each tractor looked at the last waypoint of the last machine to leave the field (to empty), that was on the same "position", as in left, center, or right.  To put that another way, I had three red tractors told to run the "left" course, the greens where on the "center" course, and the blues on "right".  It would be helpful if when a red tractor left it's staging point to continue foraging, if it pathed for a point that was the "Last Waypoint" most recently set by a red tractor (a tool running the "left" course) rather than heading for its own Last Waypoint.

It wasn't a huge matter; it still worked, it just had to drive the course without loading anything for a while.

**The second test**, 
...and the main one I'm hoping we can get a tweak for, was after regrowing the grass, forgetting the hay, and looking to do a straight up grass harvest.  BTW, if anyone is interested, this field, a quarter of the entire map, yields somewhere around 9 to 10 million liters of grass.

This time, I was determined to run this with foragers.  They have no storage capacity to speak of (20L at most), so I wondered... _Could I have them tow an auger wagon, and have the chase trailer get loaded from the auger, not the forager's pipe?_

Answer... No.  The auger wagon would behave like it wanted to unload to the chase trailer, even squirted out a little grass a few times, but the chase vehicle consistently positioned on the forager's pipe, and only the forager's pipe would consistently flow even though the auger trailer's pipe could easily reach.

**So my tweak request is something like...** 
...For a forager, can we have the pipe target the towed trailer (if there is one), and set the auger trailer's pipe as the target for the chase vehicle?  I'd also like to be able to call the next waiting chase vehicle to come and perhaps shadow the first until the first is full and peels off, then taker over as the primary chase vehicle.  Let me lay it out this way...

One of my teams of tractors are Red 1, Red 2, and Red 3.  Red 1 is at the wait point that Forager Red has set for association between CP + AD.  Red 2 and 3 are under AD control, queued and waiting to "reach" the "Wait - Red" point.  Forager Red gets started working, and calls for the vehicle at "Wait - Red".  Red 1 path finds, and heads off to sync up with Forager Red.  Red 2 moves to "Wait - Red", Red 3 queues behind it.  But with the new shadow option, Red 2 gets called to come into the field and shadow Red 1.  Red 3 moves to "Wait - Red" and idles.

Red 1 fills up, and gets passed to AD for unloading, then return (or queue) on "Wait - Red".  Red 2 moves up to start filling from Forager Red's towed auger trailer, and Red 3 gets called into the field to shadow Red 2.  Hopefully, before Red 3 takes over and gets filled up, Red 1 will have emptied, queued briefly then gotten called into the field, and be in place shadowing Red 3 to avoid delays for the Forager.

In my scenario, this is being repeated for Green team and Blue team.  There are 12 vehicles working this field.

I'm confident this would work; I successfully ran 9 machines all under CP's control most of the time.  It should handle this scenario.

So... what do y'all think.  Doable?

**Comment by Tensuko** [comment]
About your tweak:
We don't have anything to do with pipe targeting. Thats all controlled by Giants and we have no intention on forcing this to something.
Not for forage Harvester and not for auger trailer (I guess the one with auto aim pipe same as harvester).

The Unloading is done by AD for forage harvester and not CP and since AD unloading works different, there won't be a call unloader from our side.
You should ask AD on how this works with 2nd chasing unloader and all this stuff, I know they have options for this.

So conclusion is:
- No extra Pipe control from our side.
- Rest is about AD unloader.

**Reply by ADC021499** [reply]
>     * Rest is about AD unloader.

That explains... so much.  Thank you.  A whole lot of stuff just came into focus.
